firedrake/preconditioners/pmg.py

```diff
@@ -262,6 +262,35 @@ def prolongation_transfer_kernel_aij(Pk, P1):
     return kernel
 
 
+def tensor_product_space_query(V):
+    # Check if a function space V is a tensor product of
+    # either CG(N) or DG(N) on quads or hexes (same N along every direction).
+    # Additionally, return the degree N and a list of strings with the family.
+    ele = V.ufl_element()
+    cell = ele.cell()
+    isquad = cell.cellname().find("quadrilateral") >= 0  # hex == "quadrilateral * interval"
```

Comment:
`.startswith("quadrilateral")`?

Reply:
I am not sure about this. I wanted to include the case  "interval * quadrilateral"

Reply:
Ah, OK. If you're really wanting to check that the cell you have is a hypercube (I think that's what you want, since right now you would accept `quadrilateral * triangle`!), then the safe thing is to say `FIAT.reference_element.is_hypercube(V.finat_element.cell)`